aptos-move/framework/aptos-framework/sources/account.move

```diff
@@ -326,6 +360,123 @@ module aptos_framework::account {
+
+        // verify that `cap_update_table` has the correct `RotationProofChallenge` and is signed by the new private key
+        let new_auth_key = assert_valid_signature_and_get_auth_key(new_scheme, new_public_key_bytes, cap_update_table, &challenge);
+
+        // update the `OriginatingAddress` table, so we can find the originating address using the new address
+        let new_address = from_bcs::to_address(new_auth_key);
+        let address_map = &mut borrow_global_mut<OriginatingAddress>(@aptos_framework).address_map;
+        if (table::contains(address_map, curr_auth_key)) {
+            assert!(rotation_cap_offerer_address == table::remove(address_map, curr_auth_key), error::not_found(EINVALID_ORIGINATING_ADDRESS));
+        };
+        table::add(address_map, new_address, rotation_cap_offerer_address);
+
+        // update the authentication key of the offerer's account
+        let offerer_account_resource = borrow_global_mut<Account>(rotation_cap_offerer_address);
+        event::emit_event<KeyRotationEvent>(
+            &mut offerer_account_resource.key_rotation_events,
+            KeyRotationEvent {
+                old_authentication_key: offerer_account_resource.authentication_key,
+                new_authentication_key: new_auth_key,
+            }
+        );
+
+        offerer_account_resource.authentication_key = new_auth_key;
+    }
+
+    /// Offers rotation capability on behalf of `account` to the account at address `recipient_address`.
+    /// An account can delegate its rotation capability to only one other address at one time. If the account
+    /// has an existing rotation capability offer, calling this function will update the rotation capability offer with
+    /// the new `recipient_address`.
+    /// Here, `rotation_capability_sig_bytes` signature indicates that this key rotation is authorized by the account owner,
+    /// and prevents the classic "time-of-check time-of-use" attack.
+    /// For example, users usually rely on what the wallet displays to them as the transaction's outcome. Consider a contract that with 50% probability
+    /// (based on the current timestamp in Move), rotates somebody's key. The wallet might be unlucky and get an outcome where nothing is rotated,
+    /// incorrectly telling the user nothing bad will happen. But when the transaction actually gets executed, the attacker gets lucky and
+    /// the execution path triggers the account key rotation.
+    /// We prevent such attacks by asking for this extra signature authorizing the key rotation.
+    ///
+    /// @param rotation_capability_sig_bytes is the signature by the account owner's key on `RotationCapabilityOfferProofChallengeV2`.
+    /// @param account_scheme is the scheme of the account (ed25519 or multi_ed25519).
+    /// @param account_public_key_bytes is the public key of the account owner.
+    /// @param recipient_address is the address of the recipient of the rotation capability - note that if there's an existing
```

Comment:
This line ends abruptly and the line below it doesn't connect to it properly. Was this a typo or copy & paste error?

Reply:
hmmm think I pasted to the wrong line.

good catch!! thanks for reviewing this : ) I will update soon